host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1139,14 +1140,72 @@ static void update_annotation_data(RASPIVID_STATE *state)
    if (state->camera_parameters.enable_annotate & ANNOTATE_APP_TEXT)
    {
       char *text;
-      const char *refresh = raspicli_unmap_xref(state->intra_refresh_type, intra_refresh_map, intra_refresh_map_size);
 
-      asprintf(&text,  "%dk,%df,%s,%d,%s,%s",
-               state->bitrate / 1000,  state->framerate,
-               refresh ? refresh : "(none)",
-               state->intraperiod,
-               raspicli_unmap_xref(state->profile, profile_map, profile_map_size),
-               raspicli_unmap_xref(state->level, level_map, level_map_size));
+      if (state->common_settings.gps)
+      {
+         char lat[24] = {"n/a"};
+         char lon[24] = {"n/a"};
+         char alt[24] = {"Altitude n/a"};
+         char speed[24] = {"Speed n/a"};
+         char track[24] = {"Track n/a"};
+         char datetime[32] = {"Time n/a"};
+
+         struct gps_data_t *gpsdata = raspi_gps_lock();
+
+         if (gpsdata->set & TIME_SET)
+         {
+            time_t rawtime;
+            struct tm *timeinfo;
+            rawtime = gpsdata->fix.time;
+            timeinfo = localtime(&rawtime);
+            strftime(datetime, sizeof(datetime), "%Y:%m:%d %H:%M:%S", timeinfo);
+         }
+
+         if (gpsdata->online && gpsdata->fix.mode >= MODE_2D)
+         {
+            if (gpsdata->set & LATLON_SET)
+            {
+               if (!isnan(gpsdata->fix.latitude))
+                  snprintf(lat, sizeof(lat), "%.6lf", gpsdata->fix.latitude);
+
+               if (isnan(gpsdata->fix.longitude) == 0)
+                  snprintf(lon, sizeof(lon), "%.6lf", gpsdata->fix.longitude);
+            }
+
+            if ((gpsdata->set & ALTITUDE_SET) && (gpsdata->fix.mode >= MODE_3D) &&
+                  (!isnan(gpsdata->fix.altitude)))
+            {
+               snprintf(alt, sizeof(alt), "Altitude=%.2fm",
+                        (gpsdata->fix.altitude*10+0.5)/10.0 );
+            }
+
+            if ((gpsdata->set & SPEED_SET) && !isnan(gpsdata->fix.speed))
+            {
+               snprintf(speed, sizeof(speed), "Speed=%.2fms",
+                        ((gpsdata->fix.speed*MPS_TO_KPH*10+0.5)/10) );
```

Comment:
What units are we using where?
Text says milliseconds(!!!), and we're taking a speed (gps.h says in m/s), multiplying by MPS_TO_KPH, and rounding. So the text should say km/h?

Reply:
Hmm. Need to double check that. I thoughts it was m/s, but will cross check against the cgps test app.

Reply:
gps.h says 
```    double speed;	/* Speed over ground, meters/sec */```
so I think it wants to be either 
```"Speed=%.2fkm/h", ((gpsdata->fix.speed*MPS_TO_KPH*10+0.5)/10) ```
or
```"Speed=%.2fm/s", gpsdata->fix.speed) ```

Reply:
Yup, a number of those calcs can be much simplified. Another C&P thingy.